### DISCUSSION TITLE
Placeholder image consuming events meant for other views?

### DISCUSSION THREAD

**Discussion by JetForMe** [onevcat/Kingfisher#1750] [Q&A]
I'm experiencing something odd. I finally got around to experimenting with `GeometryReader` to fix the other [issue](https://github.com/onevcat/Kingfisher/discussions/1738) I was having. That seems to be working well, in that everything draws correctly.

But if the intended image doesn't load (or is nil), and only the placeholder remains, it seems that although the placeholder is correctly clipped, its bounds extend well beyond, and prevent tapping on things that it covers (our placeholder is quite tall and skinny, and we show just a portion inside).

Here's what the view looks like when the URL is nil:

<img width="455" alt="Screen Shot 2021-07-14 at 18 01 25 " src="https://user-images.githubusercontent.com/401600/125712086-d1c9042d-0426-44b0-abd5-6deb7776d218.png">

Unfortunately, you can't tap on the "Shows," "For Sale," "Lists," or "Collection" tabs. If you look at the view hierarchy in Xcode, you can see that the placeholder image extends well beyond the bounds that finally clip it when running (in preview or simulator).

![Screen Shot 2021-07-14 at 18 01 21 ](https://user-images.githubusercontent.com/401600/125712160-a9b561ef-9320-4e26-9e8a-0f4949b4a7a3.png)

Once the URL image loads, everything is fine. My code looks like this (inside of a larger view):

```swift
GeometryReader
{ geom in
    KFImage(self.thumbnailURL)
        .cancelOnDisappear(true)
        .placeholder {
            Image("image_placeholder_background")
                .resizable()
                .aspectRatio(contentMode: .fill)
                .frame(width: geom.size.width, height: geom.size.width / 0.72)      //  TODO: A bit arbitrary, this aspect ratio
        }
        .resizable()
        .aspectRatio(contentMode: .fill)
        .frame(width: geom.size.width, height: geom.size.width / 0.72)              //  TODO: A bit arbitrary, this aspect ratio
        .cornerRadius(8)
}
.aspectRatio(0.72, contentMode: .fit)
```

You can see the placeholder's image view in the hierarchy (I've selected the one on the left, highlighted in blue):

![Screen Shot 2021-07-14 at 18 18 52 ](https://user-images.githubusercontent.com/401600/125712917-25f360aa-6612-4fbe-b7bf-e90ff774d4ae.png)

I’ve tried putting `.clipped()` on various parts of this, but it has no effect.

Any idea what's causing this or how to fix it? Obviously I can shrink the placeholder image. This behavior exists on both iOS 14.4 and 15b1. I'm using Kingfisher 6.3.0

**Comment by onevcat** [comment]
:( No idea on what can cause this at all..

So it seems that you have put the `KFImage` in back of the header tab buttons, but it adds some view (the blue highlighted one) in front which covers these buttons? It beyonds my understanding of SwiftUI so I am not sure what is it at all...

In `KFImage`, the placeholder and the image should be at the [same view hierarchy level](https://github.com/onevcat/Kingfisher/blob/44450a8f564d7c0165f736ba2250649ff8d3e556/Sources/SwiftUI/KFImage.swift#L165-L196). Can it be something related to the `GeometryReader`?

By the way, do you have a sample view in which we can reproduce it easily? I also want check and see what is it. Thanks!

**Reply by JetForMe** [reply]
I don't have a sample readily available, because we're so slammed with a deadline. But I need to to show Apple. Honestly, I think this is an Apple bug. What shows in Xcode is not what renders in the sim/ on device, but the uncropped extent of the image does seem to interfere with taps. In the end I got our designer to give me a cropped version of the image, which solved the issue for now.
